### DISCUSSION TITLE
Can I set the start block number while creating a datasource template?

### DISCUSSION THREAD

**Discussion by kiroain** [graphprotocol/graph-tooling#1416] [Q&A]
The Graph supports starting a dynamically instantiated data source from a block earlier than the current block?
The dynamic data source instantiated from a template begins processing events from the block where the <Template>.create(<contract address>) function was called. Is this correct?

**Comment by itsjerryokolo** [comment]
Yes. It starts processing events from the block where the .create() function was called.

**Reply by kiroain** [reply]
Thanks for your kind reply. how can I track the past event blocks which is before the block where .create() function was called? Is it possible?

**Reply by itsjerryokolo** [reply]
You can add multiple datasources to the subgraph.yaml, and you can use the same schema for all the datasources. 
If for example you were already tracking a contract before .create() is called, you can update the entities in your schema with new data from the instantiated datasource. 

It won't sync backwards. 

Also, there is a new feature which will soon be rolled out soon. Perhaps that's what you are looking for?
https://github.com/graphprotocol/graph-node/issues/3221

https://github.com/graphprotocol/graph-node/pull/4725

**Reply by kiroain** [reply]
Thanks. That sounds impressive.
I would appreciate if you can explain more about **Polling block Handler and initialization handler** and how to use it?
